activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -183,9 +183,25 @@ def attribute_names
         \z
       /ix
 
+      using Module.new {
+        refine String do
```

Comment:
to me using a string refinement here feels a bit on the fancy side, any reason not to simply call a private method here?

Reply:
Orignally I was implemented `#fstring?` as a String extension in Active Support since a string is whether fstring or not is a concern of a string.

But @k0kubun (which is my colleague) worried about the same named API (but a bit different behavior) might be implemented in Ruby itself in the future, so I've changed AS extension to refinements.

https://github.com/rails/rails/compare/fe120e2c9aa3b0811309907dc76230cb45302720..8188298c013ceb6483bab8e3bb2a4801def61efe

https://twitter.com/k0kubun/status/1092759101182926848

Reply:
I follow I just meant use `str.object_id == (-str).object_id` in absence of an official API, then there is no need to refine string.

Reply:
`str.object_id == (-str).object_id` works as expected only for >= 2.6.0, unfortunately `-str` not always returns fstring for Ruby 2.5.

Reply:
I know all things could be implemented without refinements, but we already use that in our code base.

https://github.com/rails/rails/blob/302d739855306f9061f1f2357844cb8d76f20ae9/activesupport/lib/active_support/core_ext/enumerable.rb#L171-L176